src/texture_wrapper.rs

```diff
@@ -99,8 +178,63 @@ impl TexWrap {
         gfx: &mut Graphics,
         settings: &PersistentSettings,
         image: &DynamicImage,
+        swizzle_mask: Mat4,
+        offset_vec: Vec4,
     ) -> Option<TexWrap> {
-        Self::gen_from_dynamic_image(gfx, settings, image, Self::gen_texture_standard)
+        Self::gen_from_dynamic_image(
+            gfx,
+            settings,
+            image,
+            Self::gen_texture_standard,
+            swizzle_mask,
+            offset_vec,
+        )
+    }
+
+    fn check_union_buffer_creation(buffer_result:Result<Buffer,String>)->Buffer{
+        match buffer_result {
+            Ok(buffer) => buffer,
+            Err(error) => panic!("Problem generating union buffer: {error:?}"),
+        }
+    }
+
+    fn gen_uniform_buffer_swizzle_mask(
+        gfx: &mut Graphics,
+        swizzle_mask: Mat4,
+        offset_vec: Vec4,
+    ) -> (Buffer, Buffer) {
+        let uniform_swizzle_mask = 
+            Self::check_union_buffer_creation(
+            gfx
+            .create_uniform_buffer(1, "SwizzleMask")
+            .with_data(&swizzle_mask)
+            .build());
+
+        let uniform_offset_vector = 
+        Self::check_union_buffer_creation(gfx
+            .create_uniform_buffer(2, "OffsetVector")
+            .with_data(&offset_vec)
+            .build());
+
+        (uniform_swizzle_mask, uniform_offset_vector)
+    }
+
+    fn update_uniform_buffer(&self, gfx: &mut Graphics, swizzle_mat: Mat4, offset_vec: Vec4) {
+        gfx.set_buffer_data(&self.uniform_swizzle_mask, &swizzle_mat);
+        gfx.set_buffer_data(&self.uniform_offset_vec, &offset_vec);
+    }
+
+    fn gen_texture_background(gfx: &mut Graphics) -> Texture {
```

Comment:
this could be 
```rust
  fn gen_texture_background(gfx: &mut Graphics) -> Result<Texture> {
      let boundary_pixels_bytes: [u8; 4] = [0, 0, 0, 255];
      Ok(gfx
          .create_texture()
          .from_bytes(&boundary_pixels_bytes, 1, 1)
          .with_format(notan::app::TextureFormat::Rgba32)
          .build().map_err(|e|anyhow!("{e}"))?)
  }
```
for example.

Reply:
I did the unwrap/matching on purpose - all I need in the struct ist the texture to render later and not a texture option or a result I would have to handle more downstream.
It all boils down to one question: What to do if texture creation fails?
In my thoughts, panicking was ok at this point, what do you think?

I could have used the other function I created for "normal" textures, that's right. It was late last night :D

Reply:
No worries, I am just nitpicking!

If the error is recoverable at all, you should not panic, ever. Otherwise for the user, the app will just close and they have no idea what happened. So if it is possible at that point in time to display a warning, that should be preferable.

Reply:
Okay, I will have a look at it, when I have spare time.